spring-cloud-task-stream/src/main/java/org/springframework/cloud/task/batch/listener/EventEmittingItemProcessListener.java

```diff
@@ -64,4 +66,9 @@ else if (item.equals(result)) {
 	public void onProcessError(Object item, Exception e) {
 		messagePublisher.publishWithThrowableHeader("Exception while item was being processed", e.getMessage());
 	}
+
+	@Override
+	public int getOrder() {
+		return 0;
```

Comment:
In this case the order should be configurable for each of the event emitting listerners.  This can be done via a Spring Boot property i.e. `spring.cloud.task.events.<batch event listener>.order=<order#>` .  For example:
`spring.cloud.task.events.job-execution.order=<order#>`